micro_sam/training/sam_trainer.py

```diff
@@ -250,7 +255,14 @@ def _get_updated_points_per_mask_per_subiter(self, masks, sampled_binary_y, batc
 
             _inp["point_coords"] = updated_point_coords
             _inp["point_labels"] = updated_point_labels
-            _inp["mask_inputs"] = logits
+
+            if self.mask_prob > 0:
+                # using mask inputs for iterative prompting while training, with a probability
+                use_mask_inputs = True if random.random() < self.mask_prob else False
```

Comment:
I am not 100% sure about the order here.

Reply:
This is now fixed with the latest commit, and is good to go. Thanks!